src/test/scala/org/ergoplatform/http/routes/ApplicationApiRouteSpec.scala

```diff
@@ -0,0 +1,132 @@
+package org.ergoplatform.http.routes
+
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.Route
+import akka.http.scaladsl.testkit.{RouteTestTimeout, ScalatestRouteTest}
+import de.heikoseeberger.akkahttpcirce.FailFastCirceSupport
+import io.circe.{Decoder, Encoder, Json}
+import org.ergoplatform.ErgoBox
+import org.ergoplatform.http.api.{ApiCodecs, ApplicationApiRoute}
+import org.ergoplatform.nodeView.wallet.scanning.{ContainsScanningPredicate, ExternalAppRequest, ExternalApplication, ExternalApplicationJsonCodecs}
+import org.ergoplatform.utils.Stubs
+import org.scalatest.{FlatSpec, Matchers}
+import io.circe.syntax._
+import org.ergoplatform.http.api.ApplicationEntities.{ApplicationIdBoxId, ApplicationIdWrapper}
+import org.ergoplatform.settings.{Args, ErgoSettings}
+import org.ergoplatform.wallet.Constants.ApplicationId
+import scorex.crypto.authds.ADKey
+import scorex.utils.Random
+
+import scala.util.Try
+import scala.concurrent.duration._
+
+class ApplicationApiRouteSpec extends FlatSpec
+  with Matchers
+  with ScalatestRouteTest
+  with Stubs
+  with FailFastCirceSupport
+  with ApiCodecs {
+
+  implicit val timeout: RouteTestTimeout = RouteTestTimeout(145.seconds)
+
+  implicit val appEncoder: Encoder[ExternalApplication] = ExternalApplicationJsonCodecs.appEncoder
+  implicit val appDecoder: Decoder[ExternalApplication] = ExternalApplicationJsonCodecs.appDecoder
+
+  implicit val appReqEncoder: Encoder[ExternalAppRequest] = ExternalApplicationJsonCodecs.appReqEncoder
+  implicit val appReqDecoder: Decoder[ExternalAppRequest] = ExternalApplicationJsonCodecs.appReqDecoder
+
+  implicit val appIdWrapperEncoder: Encoder[ApplicationIdWrapper] = ApplicationIdWrapper.applicationIdWrapperEncoder
+  implicit val appIdWrapperDecoder: Decoder[ApplicationIdWrapper] = ApplicationIdWrapper.applicationIdWrapperDecoder
```

Comment:
Should we always import all this encoders individually? 
`import ExternalApplicationJsonCodecs._` doesn't work?

Reply:
made concrete imports